File: backend/src/main/java/com/java/NBE4_5_3_7/domain/interview/entity/dto/response/InterviewContentAdminResponseDto.java
package com.java.NBE4_5_3_7.domain.interview.entity.dto.response;

import com.java.NBE4_5_3_7.domain.interview.entity.InterviewCategory;
import com.java.NBE4_5_3_7.domain.interview.entity.InterviewContent;

public class InterviewContentAdminResponseDto {
    private Long id;
    private Long headId;
    private Long tailId;
    private boolean isHead;
    private boolean hasTail;
    private String keyword;
    private InterviewCategory category;
    private String question;
    private String modelAnswer;
    private Long likeCount;

    public InterviewContentAdminResponseDto() {
    }

    public InterviewContentAdminResponseDto(InterviewContent content, Long likeCount) {
        this.id = content.getInterview_content_id();
        this.headId = content.getHead_id();
        this.tailId = content.getTail_id();
        this.isHead = content.isHead();
        this.hasTail = content.isHasTail();
        this.keyword = content.getKeyword();
        this.category = content.getCategory();
        this.question = content.getQuestion();
        this.modelAnswer = content.getModelAnswer();
        this.likeCount = likeCount;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getHeadId() {
        return headId;
    }

    public void setHeadId(Long headId) {
        this.headId = headId;
    }

    public Long getTailId() {
        return tailId;
    }

    public void setTailId(Long tailId) {
        this.tailId = tailId;
    }

    public boolean isHead() {
        return isHead;
    }

    public void setHead(boolean head) {
        isHead = head;
    }

    public boolean isHasTail() {
        return hasTail;
    }

    public void setHasTail(boolean hasTail) {
        this.hasTail = hasTail;
    }

    public String getKeyword() {
        return keyword;
    }

    public void setKeyword(String keyword) {
        this.keyword = keyword;
    }

    public InterviewCategory getCategory() {
        return category;
    }

    public void setCategory(InterviewCategory category) {
        this.category = category;
    }

    public String getQuestion() {
        return question;
    }

    public void setQuestion(String question) {
        this.question = question;
    }

    public String getModelAnswer() {
        return modelAnswer;
    }

    public void setModelAnswer(String modelAnswer) {
        this.modelAnswer = modelAnswer;
    }

    public Long getLikeCount() {
        return likeCount;
    }

    public void setLikeCount(Long likeCount) {
        this.likeCount = likeCount;
    }
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/interview/entity/dto/response/InterviewContentAdminResponseDto.kt
package com.java.NBE4_5_3_7.domain.interview.entity.dto.response

import com.java.NBE4_5_3_7.domain.interview.entity.InterviewCategory
import com.java.NBE4_5_3_7.domain.interview.entity.InterviewContent

data class InterviewContentAdminResponseDto(
    var id: Long? = null,
    var headId: Long? = null,
    var tailId: Long? = null,
    var isHead: Boolean = false,
    var hasTail: Boolean = false,
    var keyword: String? = null,
    var category: InterviewCategory? = null,
    var question: String? = null,
    var modelAnswer: String? = null,
    var likeCount: Long? = null
) {
    constructor(content: InterviewContent, likeCount: Long) : this(
        id = content.interviewContentId,
        headId = content.headId,
        tailId = content.tailId,
        isHead = content.isHead,
        hasTail = content.hasTail,
        keyword = content.keyword,
        category = content.category,
        question = content.question,
        modelAnswer = content.modelAnswer,
        likeCount = likeCount
    )
}